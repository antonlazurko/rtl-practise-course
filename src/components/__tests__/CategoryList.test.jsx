import { screen } from '@testing-library/react'
import { renderWithRouter } from '../../utils/testing'
import { CategoryList } from '../CategoryList'

describe('CategoryList', () => {
    it('renders CategoryList correctly', () => {
        renderWithRouter(<CategoryList catalog={[
            {idCategory: 1, strCategory: "Beef", strCategoryThumb: "https://www.themealdb.com/images/category/beef.png", strCategoryDescription: "Test 1"},
            {idCategory: 2, strCategory: "Chicken", strCategoryThumb: "https://www.themealdb.com/images/category/chicken.png", strCategoryDescription: "Test 2"},
        ]}/>)
        expect(screen.getByRole('list')).toMatchSnapshot()
    })
    it('renders CategoryList correctly without items', () => {
        renderWithRouter(<CategoryList catalog={[]}/>)
        expect(screen.getByRole('list')).toMatchSnapshot()
    })
})