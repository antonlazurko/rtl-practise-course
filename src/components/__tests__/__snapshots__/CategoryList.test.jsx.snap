// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList renders CategoryList correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Beef"
        src="https://www.themealdb.com/images/category/beef.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Beef
      </span>
      <p>
        Test 1
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/Beef"
      >
        Watch category
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="Chicken"
        src="https://www.themealdb.com/images/category/chicken.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        Chicken
      </span>
      <p>
        Test 2
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/Chicken"
      >
        Watch category
      </a>
    </div>
  </div>
</div>
`;

exports[`CategoryList renders CategoryList correctly without items 1`] = `
<div
  class="list"
  role="list"
/>
`;
